<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default="xterm -e"/>
  <arg name="target_tracker/track"             default="openpose" />
  <arg name="target_tracker/target"            default="waving_left" />
  <arg name="target_localization/source_frame" default="/camera_top_depth_optical_frame" />
  <arg name="target_localization/target_frame" default="/base_link" />
  <arg name="under_control"                    default="false" />
  
  <!-- Bringup astra camera -->
  <include file="$(find rchomeedu_vision)/launch/multi_astra.launch" />

  <!-- Bringup dynamixel motors -->
  <include file="$(find jupiterobot_arm_bringup)/launch/jupiterobot_arm_2_3_head_bringup.launch"/>

  <!-- Config and weights folder. -->
  <rosparam file="$(find robot_vision_openvino)/config/ros.yaml" command="load" ns="openpose_ros" />
  <rosparam file="$(find robot_vision_openvino)/config/openpose.yaml" command="load" ns="openpose_ros" />
  <arg name="openpose_model/path" default="$(find robot_vision_openvino)/models/openpose" />


  <!-- Config and weights folder. -->
  <node pkg="robot_navigation_core" type="move_robot_server" name="move_robot_server" output="screen" launch-prefix="$(arg launch_prefix)"/>

  <!-- Target Localization -->
  <node pkg="robot_vision_core" type="target_localization" name="target_localization" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="target_localization/source_frame"  value="$(arg target_localization/source_frame)"/>
    <param name="target_localization/target_frame"  value="$(arg target_localization/target_frame)"/>
    <param name="under_control"                     value="$(arg under_control)" />
  </node>

  <!-- Start target tracker -->
  <node pkg="robot_vision_core" type="target_tracker" name="target_tracker" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="target_tracker/track"          value="$(arg target_tracker/track)" />
    <param name="target_tracker/target"         value="$(arg target_tracker/target)"/>
    <param name="under_control"                 value="$(arg under_control)"/>
  </node>
  
  <!-- Start openvino_openpose node -->
  <node pkg="robot_vision_openvino" type="openpose_ros" name="openpose_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="openpose_model/path"          value="$(arg openpose_model/path)" />
  </node>

</launch>
